ddl/backfilling.go

```diff
@@ -357,59 +373,65 @@ func splitTableRanges(t table.PhysicalTable, store kv.Storage, startKey, endKey
 	return ranges, nil
 }
 
-func waitTaskResults(workers []*backfillWorker, taskCnt int,
-	totalAddedCount *int64, startKey kv.Key) (kv.Key, int64, error) {
+const adjustWorkerSizeInterval = 16
+
+func waitTaskResults(scheduler *backfillScheduler, batchTasks []*reorgBackfillTask,
+	totalAddedCount *int64) (kv.Key, int64, error) {
 	var (
 		addedCount int64
-		nextKey    = startKey
 		firstErr   error
 	)
-	for i := 0; i < taskCnt; i++ {
-		worker := workers[i]
-		result := <-worker.resultCh
+	keeper := newDoneTaskKeeper(batchTasks[0].startKey)
+	taskSize := len(batchTasks)
+	for i := 0; i < taskSize; i++ {
+		result := <-scheduler.resultCh
```

Comment:
Could we add one more task into queue when received a result? If only left 3 tasks in this batch, the concurrency is not as good as beginning.

Reply:
1024 covers the tables less than 100GB. For most tables, the batch is large enough.